core-rust-bridge/src/main/java/com/radixdlt/mempool/RustMempool.java

```diff
@@ -80,7 +82,10 @@
 
 public class RustMempool implements MempoolReader<RawNotarizedTransaction> {
 
-  public RustMempool(StateManager stateManager) {
+  private final Metrics metrics;
+
+  public RustMempool(Metrics metrics, StateManager stateManager) {
+    this.metrics = Objects.requireNonNull(metrics);
```

Comment:
(minor; barely related) I think that here (and in many other applicable places) we can drop the null-checks of Guice-injected services.
As explained in https://github.com/google/guice/wiki/UseNullable#guice-forbids-null-by-default we will never actually get `null`, unless you annotate the injection point with `@Nullable` (in which case you don't want the null-check anyways!).

Reply:
Sounds good. I've removed the check from this PR, but we can leave cleaning up the remaining classes to a separate PR, I guess? Note that there are some that can be either created by guice or via a constructor (e.g. SafetyRules), where we'd probably want to keep the validation.